### DISCUSSION TITLE
$disclosure.isOpen is not working on root element

### DISCUSSION THREAD

**Discussion by jonnitto** [alpinejs/alpine#3717] [7. Bugs (archived)]
If I've have an accordion element like this:

```html
<div x-disclosure :class="$disclosure.isOpen?'open-class':'closed-class'">
    <button x-disclosure:button type="button">Button</button>
    <div x-disclosure:panel>Panel</div>
</div>
```

The class `open-class` never get's applied.

If I write it like this:

```html
<div x-disclosure :class="active?'open-class':'closed-class'" x-effect="active=$disclosure.isOpen" x-data="{active:false}">
     <button x-disclosure:button type="button">Button</button>
     <div x-disclosure:panel>Panel</div>
</div>
```

It works

**Comment by ekwoka** [comment]
hmm, it definitely should.

Does it have any errors? or anything? does the panel show/hide but just the class doesn't change?

Actually ,what is this syntax you're using here? why is it `:class={"expression"}` and not `:class="expression"`? What does the property look like in the actual Dom?

**Comment by jonnitto** [comment]
The `{`and `}` was from the template engine. I've updated the code block. I don't get any errors. The panel open and closes, but just the class doesn't change, yes.

**Comment by jonnitto** [comment]
Ok, we can close this. I had a old source without the before('bind')

**Reply by ekwoka** [reply]
That would do it
